lib/inspec/plugin/v2/installer.rb

```diff
@@ -342,13 +345,17 @@ def install_gem_to_plugins_dir(new_plugin_dependency, # rubocop: disable Metrics
 
       # Activate all current plugins before trying to activate the new one
       loader.list_managed_gems.each do |spec|
-        next if spec.name == new_plugin_dependency.name && update_mode
+        # Skip in case of update mode
+        # Skip in case using a resource pack plugin
+        next if spec.name == new_plugin_dependency.name && (update_mode || is_resource_pack_gem?(new_plugin_dependency.name))
 
         spec.activate
       end
 
       # Make sure we remove any previously loaded gem on update
-      Gem.loaded_specs.delete(new_plugin_dependency.name) if update_mode
+      # Make sure we remove any previously loaded gem when trying to use resource pack gem
+      # Resource pack gems when updated need to deactivate older version of gem
+      Gem.loaded_specs.delete(new_plugin_dependency.name) if update_mode || is_resource_pack_gem?(new_plugin_dependency.name)
```

Comment:
again here we can check in loaded specs if there are multiple versions then deactivate the ones we don't want rather than doing a regex based check

Reply:
Regex check is only for checking if a dependency is a resource pack dependency or not. And this is only called while someone uses a different version, else this is not even called here.

Reply:
ehh this will be called anytime a `install_gem_to_plugins_dir` is run